### PR TITLE
New WinForms diagnostic reference articles

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Added new diagnostic reference articles using a template from the CSharp compiler warnings area.
- Removed the obsoletion article, I don't see what value that adds. People search for specific errors and go to that page. I don't think an overview is useful.
- Reworked the TOC a bit for those messages.
  - Added the new ones to the main list broken out by error and warning.
  - Added the new ones into category sections.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [dotnet-desktop-guide/net/winforms/compiler-messages/wfo0003.md](https://github.com/dotnet/docs-desktop/blob/ea2349cce458d25c5fab88a7b7e63f0e7490c31e/dotnet-desktop-guide/net/winforms/compiler-messages/wfo0003.md) | [dotnet-desktop-guide/net/winforms/compiler-messages/wfo0003](https://review.learn.microsoft.com/en-us/dotnet/desktop/winforms/compiler-messages/wfo0003?branch=pr-en-us-1938&view=netdesktop-9.0) |
| [dotnet-desktop-guide/net/winforms/compiler-messages/wfo5001.md](https://github.com/dotnet/docs-desktop/blob/ea2349cce458d25c5fab88a7b7e63f0e7490c31e/dotnet-desktop-guide/net/winforms/compiler-messages/wfo5001.md) | [dotnet-desktop-guide/net/winforms/compiler-messages/wfo5001](https://review.learn.microsoft.com/en-us/dotnet/desktop/winforms/compiler-messages/wfo5001?branch=pr-en-us-1938&view=netdesktop-9.0) |
| [dotnet-desktop-guide/net/winforms/compiler-messages/wfo5002.md](https://github.com/dotnet/docs-desktop/blob/ea2349cce458d25c5fab88a7b7e63f0e7490c31e/dotnet-desktop-guide/net/winforms/compiler-messages/wfo5002.md) | [dotnet-desktop-guide/net/winforms/compiler-messages/wfo5002](https://review.learn.microsoft.com/en-us/dotnet/desktop/winforms/compiler-messages/wfo5002?branch=pr-en-us-1938&view=netdesktop-9.0) |
| [dotnet-desktop-guide/net/winforms/wfdev-diagnostics/wfac010.md](https://github.com/dotnet/docs-desktop/blob/ea2349cce458d25c5fab88a7b7e63f0e7490c31e/dotnet-desktop-guide/net/winforms/wfdev-diagnostics/wfac010.md) | [dotnet-desktop-guide/net/winforms/wfdev-diagnostics/wfac010](https://review.learn.microsoft.com/en-us/dotnet/desktop/winforms/wfdev-diagnostics/wfac010?branch=pr-en-us-1938&view=netdesktop-9.0) |
| [dotnet-desktop-guide/net/winforms/whats-new/net90.md](https://github.com/dotnet/docs-desktop/blob/ea2349cce458d25c5fab88a7b7e63f0e7490c31e/dotnet-desktop-guide/net/winforms/whats-new/net90.md) | [dotnet-desktop-guide/net/winforms/whats-new/net90](https://review.learn.microsoft.com/en-us/dotnet/desktop/winforms/whats-new/net90?branch=pr-en-us-1938&view=netdesktop-9.0) |

### CODE REVIEW

File: dotnet-desktop-guide/net/winforms/compiler-messages/wfo0003.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+Imports Microsoft.VisualBasic.ApplicationServices
+
+Namespace My
+    Partial Friend Class MyApplication
+        Private Sub MyApplication_ApplyApplicationDefaults(sender As Object, e As ApplyApplicationDefaultsEventArgs) Handles Me.ApplyApplicationDefaults
+            e.Font = New Font("Microsoft Sans Serif", 8.25)
+        End Sub
+    End Class
+End Namespace
+```
```
Comment (adegeo) [inline dotnet-desktop-guide/net/winforms/compiler-messages/wfo0003.md:131]: @KlausLoeffelmann This content was in the other wfac010 which seems was swapped to this warning wfo0003. I don't know why the font setting is mentioned here considering this errors shows with the DPI setting?

**Review by IEvangelist — APPROVED** [review]

File: dotnet-desktop-guide/net/winforms/compiler-messages/wfo0003.md (hunk 1 above)
Comment (KlausLoeffelmann) [reply dotnet-desktop-guide/net/winforms/compiler-messages/wfo0003.md:131]: > to this warning wfo0003

Yes. To standardize the Analyzer warning format.

I am not sure, if the numbers then are correct. We phased out the old numbers completely, and replaced them with the new format. The easiest way is to see what pops up in the compiler.

Note that the main application for this is that the _Designer_ already uses the new font settings for Design purposes. But this won't work for VB, if you set the font like described programmatically. You CAN set the font the same way as you do for C# though, but that ONLY effects the Designer at Design time, and has no effect on the runtime. The additional programmatical way is still necessary for runtime adaptation.

File: dotnet-desktop-guide/net/winforms/compiler-messages/wfo5001.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  static class Program
+  {
+      [STAThread]
+      static void Main()
+      {
+          ApplicationConfiguration.Initialize();
+  #pragma warning disable WFO5001
+          Application.SetColorMode(SystemColorMode.Dark);
+  #pragma warning restore WFO5001
+          Application.Run(new Form1());
```
Comment (KlausLoeffelmann) [inline dotnet-desktop-guide/net/winforms/compiler-messages/wfo5001.md:66]: Note, that if you change the `SystemColorMode` to `DarkMode` (or to `System`, when the System is in dark mode), that then the `SystemColors` also change to a dark mode-based palette.

File: dotnet-desktop-guide/net/winforms/compiler-messages/wfo5002.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+# Compiler Error WFO5002
+
+'System.Windows.Forms.Form.ShowAsync' is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+This compiler error is generated when using any of the following methods:
+
+- `Form.ShowAsync`
+- `Form.ShowDialogAsync`
+- `Control.InvokeAsync`
```
Comment (KlausLoeffelmann) [inline dotnet-desktop-guide/net/winforms/compiler-messages/wfo5002.md:19]: Nope. `Control.Invoke` is NOT experimental. It's final!

File: dotnet-desktop-guide/net/winforms/compiler-messages/wfo5001.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        ApplicationConfiguration.Initialize();
+        Application.SetColorMode(SystemColorMode.Dark);
+        Application.Run(new Form1());
+    }    
+}
+```
+
+## To correct this error
+
+Suppress the error and enable access to the API by either of the following methods:
```
Comment (KlausLoeffelmann) [inline dotnet-desktop-guide/net/winforms/compiler-messages/wfo5001.md:42]: Both work, but non of them is the preferred method, which is as follows:

Create a .editorConfig either on project folder level or on solution folder level, and configure the Analyzer Warnings accordingly. This, btw, is what the configuration analyzer for those warning also does:

image

**Pull request merged**